### PR TITLE
add submission 403 page

### LINKED ISSUE #894: Redesign of unauthorized submission page
Instead of having a SIGSEGV: unauthorized_access page for submissions to problems user has not solved, have a UI that informs user they are unable to view the submission as they have not solved the problem, as well as providing submission's author and problem.

### PR DESCRIPTION
**Pull request by Riolku**
Closes #894. Only shows the problem link if the problem has Solved submission access, and the problem is accessible by the user.

UI:

can see problem
can't see problem

### CODE REVIEW

File: judge/views/submission.py [hunk 1]
```diff
@@ -47,9 +53,29 @@ class SubmissionDetailBase(LoginRequiredMixin, TitleMixin, SubmissionMixin, Deta
     def get_object(self, queryset=None):
         submission = super(SubmissionDetailBase, self).get_object(queryset)
         if not submission.can_see_detail(self.request.user):
-            raise PermissionDenied()
+            raise SubmissionPermissionDenied(submission)
         return submission
 
+    def get(self, request, *args, **kwargs):
+        try:
+            return super(SubmissionDetailBase, self).get(request, *args, **kwargs)
```
Comment (Ninjaclasher) [inline judge/views/submission.py:61]:
```suggestion
            return super().get(request, *args, **kwargs)
```
This is Python 2 syntax. We're slowly migrating to Python 3 syntax, so any new code should just use `super()`.

Comment (Riolku) [reply judge/views/submission.py]: Alright, pushed.

File: judge/views/submission.py [hunk 2]
```diff
@@ -37,6 +38,11 @@ def submission_related(queryset):
         .prefetch_related('contest_object__authors', 'contest_object__curators')
 
 
+class SubmissionPermissionDenied(Exception):
```
Comment (quantum5) [inline judge/views/submission.py:41]: Would it make more sense semantically to inherit from `PermissionDenied`?

**Review by quantum5 — COMMENTED** [review]
Otherwise LGTM.

File: judge/views/submission.py (hunk 2 above)
Comment (Riolku) [reply judge/views/submission.py]: Yes, it definitely would, updated.

**Review by quantum5 — APPROVED** [review]

**Pull request merged**